Install the BoostComputeConfig to /usr/lib/cmake instead of /usr/share/cmake.

Patch by Marius Brehler.
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -81,7 +81,7 @@ configure_file(
 # install cmake config file
 install(
   FILES ${BoostCompute_BINARY_DIR}/BoostComputeConfig.cmake
-  DESTINATION share/cmake/BoostCompute
+  DESTINATION lib/cmake/BoostCompute
 )
 
 # install header files
